lib/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -387,4 +388,86 @@ public boolean isVcSuccessful(List<VcStatusDto> currentVcStatuses, String criIss
+
+            if (isVcSuccessful(currentVcStatuses, componentId)) {
+                try {
+                    JsonNode nameNode =
+                            objectMapper
+                                    .readTree(
+                                            SignedJWT.parse(item.getCredential())
+                                                    .getPayload()
+                                                    .toString())
+                                    .path(VC_CLAIM)
+                                    .path(VC_CREDENTIAL_SUBJECT)
+                                    .path(NAME_PROPERTY_NAME);
+
+                    if (nameNode.isMissingNode()) {
+                        LOGGER.error("Name property is missing from passport VC");
+                        throw new HttpResponseExceptionWithErrorBody(
+                                500, ErrorResponse.FAILED_TO_GENERATE_IDENTIY_CLAIM);
+                    }
+
+                    JsonNode birthDateNode =
+                            objectMapper
+                                    .readTree(
+                                            SignedJWT.parse(item.getCredential())
+                                                    .getPayload()
+                                                    .toString())
+                                    .path(VC_CLAIM)
+                                    .path(VC_CREDENTIAL_SUBJECT)
+                                    .path(BIRTH_DATE_PROPERTY_NAME);
+
+                    if (birthDateNode.isMissingNode()) {
+                        LOGGER.error("BirthDate property is missing from passport VC");
+                        throw new HttpResponseExceptionWithErrorBody(
+                                500, ErrorResponse.FAILED_TO_GENERATE_IDENTIY_CLAIM);
+                    }
+
+                    List<Name> names =
+                            objectMapper.treeToValue(
+                                    nameNode,
+                                    objectMapper
+                                            .getTypeFactory()
+                                            .constructCollectionType(List.class, Name.class));
+                    List<BirthDate> birthDates =
+                            objectMapper.treeToValue(
+                                    birthDateNode,
+                                    objectMapper
+                                            .getTypeFactory()
+                                            .constructCollectionType(List.class, BirthDate.class));
+
+                    identityClaims.add(new IdentityClaim(names, birthDates));
+                } catch (ParseException | JsonProcessingException e) {
+                    LOGGER.error("Failed to parse VC JWT because: {}", e.getMessage());
+                    throw new HttpResponseExceptionWithErrorBody(
+                            500, ErrorResponse.FAILED_TO_GENERATE_IDENTIY_CLAIM);
```

Comment:
Think we need reorganise this - sonar is complaining as this block is duplicated from the generateUserIdentity() method and I don't think we should be throwing this error in the context of name correlation